projects/go-lib/src/lib/components/go-datepicker/go-calendar.component.ts

```diff
@@ -121,5 +122,25 @@ export class GoCalendarComponent implements OnDestroy, OnInit {
     }
     this.currentMonth = this.selectedDate.getMonth();
     this.updateYear(this.selectedDate.getFullYear());
+    if (this.appendTo === 'body') {
+      setTimeout(() => {
```

Comment:
Does this need to be in a `setTimeout`?

Reply:
I think this setTimeout is required otherwise viewchild property is coming as an **undefined**. Because viewChild will work after dom load. For that reason, I have added that setTimeout.

Reply:
I've added some further comments, but this `setTimeout` can be removed with the `ViewChild` approach. Review my other comments, you'll need to make a couple adjustments to get it working properly.